cmake/sysbuild/image_signing_split.cmake

```diff
@@ -0,0 +1,256 @@
+  endif()
+
+  if(NOT "${keyfile}" STREQUAL "")
+    set(imgtool_extra -k "${keyfile}" ${imgtool_extra})
+  endif()
+
+  set(imgtool_args ${imgtool_extra})
+
+  # Extensionless prefix of any output file.
+  set(output_internal ${ZEPHYR_BINARY_DIR}/${KERNEL_NAME}.internal.signed)
+  set(output_external ${ZEPHYR_BINARY_DIR}/${KERNEL_NAME}.external.signed)
+  set(output_merged ${ZEPHYR_BINARY_DIR}/${KERNEL_NAME}.signed)
+
+  if(CONFIG_BUILD_WITH_TFM)
+#    set(input ${APPLICATION_BINARY_DIR}/zephyr/tfm_merged)
+    message(FATAL_ERROR "TFM is not currently supported with QSPI XIP")
+  else()
+    set(input ${APPLICATION_BINARY_DIR}/zephyr/${KERNEL_NAME})
+    set(input_internal ${ZEPHYR_BINARY_DIR}/${KERNEL_NAME}.internal)
+    set(input_external ${ZEPHYR_BINARY_DIR}/${KERNEL_NAME}.external)
+  endif()
+
+  # List of additional build byproducts.
+  set(byproducts)
+
+  # 'west sign' arguments for confirmed, unconfirmed and encrypted images.
+  set(unconfirmed_args)
+  set(confirmed_args)
+  set(encrypted_args)
+
+  # Split files apart
+  split(
+    ELF_FILE_IN ${ZEPHYR_BINARY_DIR}/${KERNEL_ELF_NAME}
+    ELF_SECTION_NAMES ".extflash_text_reloc;.extflash_rodata_reloc"
+    HEX_INCLUDE_FILE_OUT ${input_internal}.hex
+    HEX_EXCLUDE_FILE_OUT ${input_external}.hex
+  )
+
+  # Set up .hex outputs.
+  if(CONFIG_BUILD_OUTPUT_HEX)
+    set(unconfirmed_internal_args ${input_internal}.hex ${output_internal}.hex)
+    set(unconfirmed_external_args ${input_external}.hex ${output_external}.hex)
+    list(APPEND byproducts ${output_internal}.hex ${output_external}.hex)
+
+    # Do not run zephyr_runner_file here as PM will provide the merged hex file from
+    # sysbuild's scope unless this is a variant image
+    if(CONFIG_NCS_IS_VARIANT_IMAGE)
+      zephyr_runner_file(hex ${output_merged}.hex)
+    endif()
+
+    set(BYPRODUCT_KERNEL_SIGNED_HEX_NAME "${output_merged}.hex"
+        CACHE FILEPATH "Signed kernel hex file" FORCE
+    )
+
+    # Add the west sign calls and their byproducts to the post-processing
+    # steps for zephyr.elf.
+    #
+    # CMake guarantees that multiple COMMANDs given to
+    # add_custom_command() are run in order, so adding the 'west sign'
+    # calls to the "extra_post_build_commands" property ensures they run
+    # after the commands which generate the unsigned versions.
+    set_property(GLOBAL APPEND PROPERTY extra_post_build_commands COMMAND
+      ${imgtool_internal_sign} ${imgtool_args} ${unconfirmed_internal_args})
+    set_property(GLOBAL APPEND PROPERTY extra_post_build_commands COMMAND
+      ${imgtool_external_sign} ${imgtool_args} ${unconfirmed_external_args})
+
+    # Combine the signed hex files into a single output hex file
+    set_property(GLOBAL APPEND PROPERTY extra_post_build_commands COMMAND
+      ${PYTHON_EXECUTABLE} ${ZEPHYR_BASE}/scripts/build/mergehex.py -o ${output_merged}.hex ${output_internal}.hex ${output_external}.hex)
+
+    if(NOT "${keyfile_enc}" STREQUAL "")
```

Comment:
The build system is producing encrypted and unencrypted image at once. 
I see that is common pattern in NCS for sysbuild. This might be misleading for end-user. I would expect only encrypted image in case encryption is "ON".
Of course this is probably documented, Maybe I'm too UX sensible.

Reply:
This is the same as it is in zephyr